[Filebeat] Improve ASA/FTD pipeline

hitchatwork opened the pull request:
<!-- Type of change
Please label this PR with one of the following labels, depending on the scope of your change:
- Enhancement
-->

## What does this PR do?

<!-- Mandatory
-->
Fixes #21658

For messages 716002:
-Changed to GROK; allows for better parsing of event.reason
-Added field for cisco.webvpn.group.name
-Added field for event.reason per cisco docs for why session was terminated

For messages 722051:
-Add angle brackets to dissect to properly dissect the message, per cisco docs
-Added field for cisco.webvpn.group.name

For messages 305011:
-Change to GROK; allows for variance in message format with identity firewall

For messages 302013/302015:
-Change to GROK; allows for variance in message format with identity firewall

For messages 302014/302016/302021:
-Added patterns and modified order of patterns of GROK to better match teardown messages
-Note that order of processing is important as the most specific messages are matched first, falling through to the appropriate match.
-Added temp fields for teardown initiator and user; defined in cisco docs but currently no real place to put them, but could be in future.
-Added icmp.request.type and icmp.request.code parsing for 302021 messages
-Changed typecast to 'long' for network.bytes; it is poosible this value exceeds integer range, and ECS reference field is defined as long. This prevents processor failures (spotted in live data set when value is greater than int; changing typecast fixed).
-Changed duration matching from TIME to NOTSPACE, as long-lived connections (over 24 hours) don't match TIME.

Additionally, added descriptions to many fields to make them easier to find in 7.9 Kibana Ingest Node Pipeline editor.

## Why is it important?

<!-- Mandatory
-->
Improved parsing of ASA/FTD messages mentioned above.

## Checklist

<!-- Mandatory
Add a checklist of things that are required to be reviewed in order to have the PR approved

List here all the items you have verified BEFORE sending this PR. Please DO NOT remove any item, striking through those that do not apply. (Just in case, strikethrough uses two tildes. ~~Scratch this.~~)
-->

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

<!-- Recommended
Add a checklist of things that are required to be reviewed in order to have the PR approved
-->
- [ ]

## How to test this PR locally

<!-- Recommended
-->
Test against ASA and FTD logs; can be provided if needed.

## Related issues

<!-- Recommended
Link related issues below. Insert the issue link or reference after the word "Closes" if merging this should automatically close it.
-->
- Closes #21658

elasticmachine commented:
Since this is a community submitted pull request, a Jenkins build has not been kicked off automatically. Can an Elastic organization member please verify the contents of this patch and then kick off a build manually?

elasticmachine commented:
Since this is a community submitted pull request, a Jenkins build has not been kicked off automatically. Can an Elastic organization member please verify the contents of this patch and then kick off a build manually?

elasticmachine commented:
## :grey_exclamation: Build Aborted
  
  > The PR is not allowed to run in the CI yet
  


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-22001/detail/PR-22001/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-22001/detail/PR-22001/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-22001/detail/PR-22001/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-22001/detail/PR-22001/1//artifacts) 

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats

* Build Cause: `[Pull request #22001 opened]`

* Reason: `The PR is not allowed to run in the CI yet`

* Start Time: 2020-10-20T06:55:13.786+0000

* Duration: 4 min 26 sec


* Commit: 9e6df4070eea08594852c27856e387b58e592a82




</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY-->


### Steps errors [![2](https://img.shields.io/badge/2%20-red)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-22001/detail/PR-22001/1//pipeline)
  <details><summary>Expand to view the steps failures</summary>
  <p>
  
  
  * **Name**: `Error signal`
    * Description: githubPrCheckApproved: The PR is not allowed to run in the CI yet. (Only users with write permission
    
    * Duration: 0 min 0 sec
    
    * Start Time: 2020-10-20T06:59:34.894+0000
    
    * [log](https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/PR-22001/runs/1/steps/77/log)
    
  
  * **Name**: `General Build Step`
    * Description: 
    
    * Duration: 0 min 1 sec
    
    * Start Time: 2020-10-20T06:59:38.019+0000
    
    * [log](https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/PR-22001/runs/1/steps/100/log)
    
  
  </p>
  </details>



<!-- LOG OUTPUT-->
### Log output

<details><summary>Expand to view the last 100 lines of log output</summary>
<p>

```
[2020-10-20T06:58:08.529Z]  > git --version # 'git version 2.17.1'
[2020-10-20T06:58:08.529Z] using GIT_SSH to set credentials GitHub user @elasticmachine SSH key
[2020-10-20T06:58:08.535Z]  > git fetch --no-tags --progress -- git@github.com:elastic/beats.git +refs/heads/*:refs/remotes/origin/* # timeout=15
[2020-10-20T06:58:29.870Z] Cleaning workspace
[2020-10-20T06:58:29.894Z] Using shallow fetch with depth 10
[2020-10-20T06:58:29.894Z] Pruning obsolete local branches
[2020-10-20T06:58:29.844Z]  > git config remote.origin.url git@github.com:elastic/beats.git # timeout=10
[2020-10-20T06:58:29.847Z]  > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
[2020-10-20T06:58:29.859Z]  > git config remote.origin.url git@github.com:elastic/beats.git # timeout=10
[2020-10-20T06:58:29.872Z]  > git rev-parse --verify HEAD # timeout=10
[2020-10-20T06:58:29.878Z] No valid HEAD. Skipping the resetting
[2020-10-20T06:58:29.878Z]  > git clean -fdx # timeout=10
[2020-10-20T06:58:29.896Z] Fetching upstream changes from git@github.com:elastic/beats.git
[2020-10-20T06:58:29.896Z] using GIT_SSH to set credentials GitHub user @elasticmachine SSH key
[2020-10-20T06:58:30.992Z] Merging remotes/origin/master commit 6955665227cb13504c52a01d3cbccd0a28c7ed9e into PR head commit 9e6df4070eea08594852c27856e387b58e592a82
[2020-10-20T06:58:29.900Z]  > git fetch --no-tags --progress --prune -- git@github.com:elastic/beats.git +refs/pull/22001/head:refs/remotes/origin/PR-22001 +refs/heads/master:refs/remotes/origin/master # timeout=15
[2020-10-20T06:58:30.994Z]  > git config core.sparsecheckout # timeout=10
[2020-10-20T06:58:30.997Z]  > git checkout -f 9e6df4070eea08594852c27856e387b58e592a82 # timeout=15
[2020-10-20T06:58:32.451Z] Merge succeeded, producing 9e6df4070eea08594852c27856e387b58e592a82
[2020-10-20T06:58:32.451Z] Checking out Revision 9e6df4070eea08594852c27856e387b58e592a82 (PR-22001)
[2020-10-20T06:58:33.201Z] Commit message: "Update asa-ftd-pipeline.yml"
[2020-10-20T06:58:33.201Z] Cleaning workspace
[2020-10-20T06:58:33.358Z]  > git --version # timeout=10
[2020-10-20T06:58:33.369Z]  > git --version # 'git version 2.17.1'
[2020-10-20T06:58:33.373Z] fatal: bad object 78a4cef649f44f7bf375ad0a4750b542218f02b8
[2020-10-20T06:58:32.422Z]  > git remote # timeout=10
[2020-10-20T06:58:32.426Z]  > git config --get remote.origin.url # timeout=10
[2020-10-20T06:58:32.430Z] using GIT_SSH to set credentials GitHub user @elasticmachine SSH key
[2020-10-20T06:58:32.434Z]  > git merge 6955665227cb13504c52a01d3cbccd0a28c7ed9e # timeout=10
[2020-10-20T06:58:32.445Z]  > git rev-parse HEAD^{commit} # timeout=10
[2020-10-20T06:58:32.453Z]  > git config core.sparsecheckout # timeout=10
[2020-10-20T06:58:32.462Z]  > git checkout -f 9e6df4070eea08594852c27856e387b58e592a82 # timeout=15
[2020-10-20T06:58:33.204Z]  > git rev-parse --verify HEAD # timeout=10
[2020-10-20T06:58:33.212Z] Resetting working tree
[2020-10-20T06:58:33.212Z]  > git reset --hard # timeout=10
[2020-10-20T06:58:33.288Z]  > git clean -fdx # timeout=10
[2020-10-20T06:58:34.717Z] Masking supported pattern matches of $GIT_USERNAME or $GIT_PASSWORD
[2020-10-20T06:58:35.459Z] + git fetch https://****:****@github.com/elastic/beats.git +refs/pull/*/head:refs/remotes/origin/pr/*
[2020-10-20T06:59:31.812Z] Archiving artifacts
[2020-10-20T06:59:32.557Z] + git rev-parse HEAD
[2020-10-20T06:59:32.929Z] + git rev-parse HEAD
[2020-10-20T06:59:33.256Z] + git rev-parse origin/pr/22001
[2020-10-20T06:59:33.305Z] [INFO] githubEnv: Found Git Build Cause: pr
[2020-10-20T06:59:33.637Z] Masking supported pattern matches of $GITHUB_TOKEN
[2020-10-20T06:59:34.523Z] [WARN] githubApiCall: The REST API call https://api.github.com/repos/elastic/beats/pulls/22001/reviews return 0 elements
[2020-10-20T06:59:34.576Z] [INFO] githubPrCheckApproved: Title: Update asa-ftd-pipeline.yml - User: hitchatwork - Author Association: FIRST_TIME_CONTRIBUTOR
[2020-10-20T06:59:34.913Z] ERROR: githubPrCheckApproved: The PR is not allowed to run in the CI yet
[2020-10-20T06:59:34.913Z] ERROR: githubPrCheckApproved: The PR is not allowed to run in the CI yet. (Only users with write permissions can do so.)
[2020-10-20T06:59:34.992Z] [INFO] Let's stop build #1. The PR is not allowed to run in the CI yet
[2020-10-20T06:59:35.012Z] Sleeping for 5 sec
[2020-10-20T06:59:36.212Z] Stage "Lint" skipped due to earlier failure(s)
[2020-10-20T06:59:36.285Z] Stage "Build&Test" skipped due to earlier failure(s)
[2020-10-20T06:59:37.992Z] [INFO] unstashV2: JOB_GCS_BUCKET is set. bucket param got precedency instead.
[2020-10-20T06:59:38.013Z] [INFO] unstashV2: JOB_GCS_CREDENTIALS is set. credentialsId param got precedency instead.
[2020-10-20T06:59:38.158Z] Error when executing always post condition:
[2020-10-20T06:59:38.166Z] com.google.api.client.googleapis.json.GoogleJsonResponseException: 404 Not Found
[2020-10-20T06:59:38.166Z] {
[2020-10-20T06:59:38.166Z]   "code" : 404,
[2020-10-20T06:59:38.166Z]   "errors" : [ {
[2020-10-20T06:59:38.166Z]     "domain" : "global",
[2020-10-20T06:59:38.166Z]     "message" : "No such object: beats-ci-temp/Beats/beats/PR-22001-1/source/source.tgz",
[2020-10-20T06:59:38.166Z]     "reason" : "notFound"
[2020-10-20T06:59:38.166Z]   } ],
[2020-10-20T06:59:38.166Z]   "message" : "No such object: beats-ci-temp/Beats/beats/PR-22001-1/source/source.tgz"
[2020-10-20T06:59:38.166Z] }
[2020-10-20T06:59:38.166Z] 	at com.google.api.client.googleapis.json.GoogleJsonResponseException.from(GoogleJsonResponseException.java:150)
[2020-10-20T06:59:38.166Z] 	at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:113)
[2020-10-20T06:59:38.166Z] 	at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:40)
[2020-10-20T06:59:38.166Z] 	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest$1.interceptResponse(AbstractGoogleClientRequest.java:321)
[2020-10-20T06:59:38.166Z] 	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1056)
[2020-10-20T06:59:38.166Z] 	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:419)
[2020-10-20T06:59:38.166Z] 	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:352)
[2020-10-20T06:59:38.166Z] 	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:469)
[2020-10-20T06:59:38.166Z] 	at com.google.jenkins.plugins.util.RequestCallable$1.call(RequestCallable.java:46)
[2020-10-20T06:59:38.166Z] 	at com.google.jenkins.plugins.util.Executor$Default.execute(Executor.java:125)
[2020-10-20T06:59:38.166Z] Caused: com.google.jenkins.plugins.util.NotFoundException
[2020-10-20T06:59:38.166Z] 	at com.google.jenkins.plugins.util.Executor$Default.execute(Executor.java:132)
[2020-10-20T06:59:38.166Z] 	at com.google.jenkins.plugins.util.Executor.execute(Executor.java:49)
[2020-10-20T06:59:38.166Z] 	at com.google.jenkins.plugins.storage.DownloadStep.resolveBucketPath(DownloadStep.java:419)
[2020-10-20T06:59:38.166Z] 	at com.google.jenkins.plugins.storage.DownloadStep.perform(DownloadStep.java:210)
[2020-10-20T06:59:38.166Z] Caused: java.io.IOException: Exception while performing download
[2020-10-20T06:59:38.166Z] 	at com.google.jenkins.plugins.storage.DownloadStep.perform(DownloadStep.java:223)
[2020-10-20T06:59:38.166Z] 	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:80)
[2020-10-20T06:59:38.166Z] 	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:67)
[2020-10-20T06:59:38.166Z] 	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
[2020-10-20T06:59:38.166Z] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[2020-10-20T06:59:38.166Z] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[2020-10-20T06:59:38.166Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2020-10-20T06:59:38.166Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2020-10-20T06:59:38.166Z] 	at java.lang.Thread.run(Thread.java:748)
[2020-10-20T06:59:38.166Z] 
[2020-10-20T06:59:38.598Z] Running on Jenkins in /var/lib/jenkins/workspace/Beats_beats_PR-22001
[2020-10-20T06:59:38.865Z] [INFO] getVaultSecret: Getting secrets
[2020-10-20T06:59:38.920Z] Masking supported pattern matches of $VAULT_ADDR or $VAULT_ROLE_ID or $VAULT_SECRET_ID
[2020-10-20T06:59:39.759Z] + chmod 755 generate-build-data.sh
[2020-10-20T06:59:39.759Z] + ./generate-build-data.sh https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-22001/ https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-22001/runs/1 ABORTED 265703
[2020-10-20T06:59:39.759Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-22001/runs/1/steps/?limit=10000 -o steps-info.json
[2020-10-20T06:59:40.670Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-22001/runs/1/tests/?status=FAILED -o tests-errors.json
[2020-10-20T06:59:40.670Z] Retry 1/3 exited 22, retrying in 1 seconds...
[2020-10-20T06:59:42.014Z] Retry 2/3 exited 22, retrying in 2 seconds...

```

</p>
</details>



<!--COMMENT_GENERATED_WITH_ID_comment.id-->

hitchatwork commented:
After latest CI build attempt, the event.reason field is being flagged as not previously defined. With the enhanced grok'ing, this is useful, especially to SIEM investigators.

Also, added fields for icmp types [again, flagged as previously not defined] when present that match up with packetbeat's definitions, which also can be useful to SIEM users.

andrewstucki commented:
Hi @hitchatwork thank you for the contribution, I haven't yet taken a look at the actual pipeline changes yet, but from a quick glance at the build in Travis, you're hitting a number of

```
Exception: Key '*' found in event is not documented!
```

What those are pointing to is that you'll need to add the specific additional fields that are now filled in the corresponding `fields.yml` files that are used to generate the Elasticsearch index mapping. For the cisco `asa` module that should be under `x-pack/filebeat/module/cisco/asa/_meta/fields.yml`

hitchatwork commented:
Hi @andrewstucki - thanks for the pointer on updating those files.  I updated the ASA and FTD meta fields.yml files, but now travis has thrown:
"Error: some files are not up-to-date. Run 'mage fmt update' then review and commit the changes. Modified: [filebeat/docs/fields.asciidoc]"

Not sure what to do here.  Any advice appreciated.

P1llus commented:
@hitchatwork if you run "mage update" in `./beats/x-pack/filebeat` and then again in `./beats/filebeat` you should see some new files being changed from the output of `git status`, please add those as well.

It complains that when you update fields.yml, the updates do not exist in the overall field definition for filebeat, which is done through mage update.

Hope that helps! :)

hitchatwork commented:
> @hitchatwork if you run "mage update" in `./beats/x-pack/filebeat` and then again in `./beats/filebeat` you should see some new files being changed from the output of `git status`, please add those as well.
> 
> It complains that when you update fields.yml, the updates do not exist in the overall field definition for filebeat, which is done through mage update.
> 
> Hope that helps! :)

Mage keeps complaining about files not found, and it appears the magefile references a 'v7' folder structure that isn't present:
Error parsing magefiles: error running "go list -f {{.Dir}}||{{.Name}} github.com/elastic/beats/v7/filebeat/scripts/mage/generate": exit status 1
can't load package: package github.com/elastic/beats/v7/filebeat/scripts/mage/generate: cannot find package "github.com/elastic/beats/v7/filebeat/scripts/mage/generate" in any of:

I checked on github and in my repositories and I don't see a v7 anywhere.

Thoughts?

kvch commented:
`v7` is part of the import path of Beats, as we are currently in the v7 major release. It is not reflected in the folder structure.

What is in your `GOROOT`, `GOPATH` env variable and `go.mod` file?

hitchatwork commented:
@kvch Appreciate your help here.  Answers to your questions:

I'm using Ubuntu Server 18.04

GOPATH="/home/hitch/"
GOROOT="/usr/lib/go-1.10"

The binary is at /usr/bin/go

I cloned the beats repository to:
/home/hitch/src/github.com/elastic/beats

For go.mod:
hitch@server:~/src/github.com/magefile/mage$ cat go.mod 
module github.com/magefile/mage

go 1.12
hitch@server:~/src/github.com/magefile/mage$

hitchatwork commented:
@kvch So I had an outdated version of go that was causing my initial problems.

Now I'm able to run mage update, but when I do that in the beats/filebeat directory, it throws this error at the end:
```
warn: failed to upgrade pip (ignoring): failed to run "/home/hitch/src/github.com/hitchatwork/beats/build/ve/linux/bin/pip install -U pip: fork/exec /home/hitch/src/github.com/hitchatwork/beats/build/ve/linux/bin/pip: no such file or directory"Error: failed to run "/home/hitch/src/github.com/hitchatwork/beats/build/ve/linux/bin/pip install -U wheel: fork/exec /home/hitch/src/github.com/hitchatwork/beats/build/ve/linux/bin/pip: no such file or directory"
hitch@elk-02:~/src/github.com/hitchatwork/beats/filebeat$
```

Go env:
```
GOPATH="/home/hitch/go"
GOROOT="/usr/local/go"
```

I also searched a bit, and tried deleting the build/ve directory.  But same error.  I checked to make sure I have python3 and the venv module, which I do:
```
hitch@elk-02:~/src/github.com/hitchatwork/beats/filebeat$ python3
Python 3.6.9 (default, Oct  8 2020, 12:12:24) 
[GCC 8.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import venv
>>> 
```
Also, I am running Ubuntu Server 18.04

jsoriano commented:
@hitchatwork do you have the `python3-venv`, or `python3.6-venv`, packages installed? `import venv` is not enough check because this always work in python 3, this package includes some scripts that are needed to actually create virtual environments.

If you have it, try to remove the virtual environments (the `/home/hitch/src/github.com/hitchatwork/beats/build/ve` directory) and run `mage update` again.

kvch commented:
jenkins run tests

elasticmachine commented:
## :bug: Flaky test report
:grey_exclamation: There are test failures but not known flaky tests.


<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

### Test stats :test_tube:
| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 8  |
| Passed       | 4467  |
| Skipped      | 561 |
| Total        | 5036   |



### Genuine test errors [![8](https://img.shields.io/badge/8%20-red)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-22001/detail/PR-22001/12//tests)
:broken_heart: There are test failures but not known flaky tests, most likely a genuine test failure.

        * **Name**: `Build&Test / x-pack/filebeat-build / test_fileset_file_078_cisco – x-pack.filebeat.tests.system.test_xpack_modules.XPackTest`
        * **Name**: `Build&Test / x-pack/filebeat-build / test_fileset_file_079_cisco – x-pack.filebeat.tests.system.test_xpack_modules.XPackTest`
        * **Name**: `Build&Test / x-pack/filebeat-build / test_fileset_file_081_cisco – x-pack.filebeat.tests.system.test_xpack_modules.XPackTest`
        * **Name**: `Build&Test / x-pack/filebeat-build / test_fileset_file_084_cisco – x-pack.filebeat.tests.system.test_xpack_modules.XPackTest`
        * **Name**: `Build&Test / x-pack/filebeat-build / test_fileset_file_085_cisco – x-pack.filebeat.tests.system.test_xpack_modules.XPackTest`
        * **Name**: `Build&Test / x-pack/filebeat-build / test_fileset_file_088_cisco – x-pack.filebeat.tests.system.test_xpack_modules.XPackTest`
        * **Name**: `Build&Test / x-pack/filebeat-build / test_fileset_file_089_cisco – x-pack.filebeat.tests.system.test_xpack_modules.XPackTest`
        * **Name**: `Build&Test / x-pack/filebeat-build / test_fileset_file_091_cisco – x-pack.filebeat.tests.system.test_xpack_modules.XPackTest`


</p>
</details>

<!--COMMENT_GENERATED_WITH_ID_flaky.id-->

hitchatwork commented:
@kvch and @jsoriano - thanks for the pointers.  Was finally able to get all the dependencies/env setup.

Now the only issues I have from the Travis build are:
1) Dictionary items added or removed (not sure how to fix this)
2) Grok patterns not matching but I believe the test files have invalid tests patterns (I opened issues #22042 and #22056 that identify what I believe the problems with the test files are)
3) Dictionary items changed - where the username contained the open and closing parentheses; I cleaned this up with the grok expressions since the username shouldn't actually contain the parentheses anyways.

Thanks!

hitchatwork commented:
Swill awaiting a reviewer and/or tips fixing dictionary keys mentioned in my last comment/question.

Thanks!
👍

jsoriano commented:
@hitchatwork oops, sorry for the long delay, could you please update your branch with master before continuing? Thanks!

hitchatwork commented:
> @hitchatwork oops, sorry for the long delay, could you please update your branch with master before continuing? Thanks!

Updated branch with master.

jsoriano commented:
jenkins run the tests please

jsoriano commented:
Linting issue doesn't seem related now, let's see if https://github.com/elastic/beats/pull/23313 fixes this...

jsoriano commented:
@hitchatwork sorry, could you update the branch with master again?

hitchatwork commented:
@jsoriano no worries - just updated again.

jsoriano commented:
jenkins run the tests again please

jsoriano commented:
@hitchatwork I think that the pending errors are caused by your changes in the pipeline. This is not necessarily bad :slightly_smiling_face: but you will need to regenerate the rest of test files in the module.

For that, run the following command from the `x-pack/filebeat` directory:
```
GENERATE=1 TESTING_FILEBEAT_MODULES=cisco mage pythonIntegTest
```
(it can take a while)

If after that you still have errors, you may need to review your changes in the pipeline.

You can also try to fix the IP addresses you reported in https://github.com/elastic/beats/issues/22042, and regenerate with the previous command.

@adriansr do you know if these 5-octet "IPs" reported in https://github.com/elastic/beats/issues/22042 are correct? could they be changed to normal IPs or does this fifth number represent some other thing?

hitchatwork commented:
@jsoriano Thanks for that pointer.  All but the two tests with the funky IP address passed.

I did fix those log message lines and posted a PR #23331 (but that will need you or someone's review there too).  Might be worth reviewing/posting that first, and then kicking this one off again (since the test pulls the sample logs from the elastic repo and not mine).

jsoriano commented:
jenkins run the tests please

jsoriano commented:
@hitchatwork #23331 has been merged, could you update this branch?

hitchatwork commented:
@jsoriano can you kick jenkins off again now that the sample log files have been fixed?

P1llus commented:
jenkins run the tests please

jsoriano commented:
jenkins run the tests again please

elasticmachine commented:
Pinging @elastic/ingest-management (Team:Ingest Management)

kuisathaverat reviewed (changes requested):
there are changes in Jenkinsfile.xml files and should not

kuisathaverat commented on dev-tools/make/mage-install.mk:
```diff
@@ -1,4 +1,4 @@
-MAGE_VERSION     ?= v1.10.0
+MAGE_VERSION     ?= v1.11.0
```
mage was update yesterday.

hitchatwork commented:
@kuisathaverat - I think those changes were from rebasing from master.  The main file changes I ended up making were to:
x-pack/filebeat/module/cisco/shared/ingest/asa-ftd-pipeline.yml
x-pack/filebeat/module/cisco/asa/_meta/fields.yml
x-pack/filebeat/module/cisco/ftd/_meta/fields.yml
x-pack/filebeat/module/cisco/fields.go

The test files got updated too.

As of this writing, all tests passed locally.

If you, or @jsoriano or someone else can kick off jenkins, it should pass and should be reviewable by relevant parties.

Thanks!

jsoriano commented:
@hitchatwork your branch contains several commits that are not related to your change, see https://github.com/elastic/beats/pull/22001/commits, they seem to be duplicated from master.

This is making the review a bit complicated, because it is hard to say if this would change something in an unexpected way. Could you please try to clean your branch so it includes only your changes? Feel free to create a new PR if you find it easier to copy your changes into a new branch.

Let me know if you need help with this.

hitchatwork commented:
> @hitchatwork your branch contains several commits that are not related to your change, see https://github.com/elastic/beats/pull/22001/commits, they seem to be duplicated from master.
> 
> This is making the review a bit complicated, because it is hard to say if this would change something in an unexpected way. Could you please try to clean your branch so it includes only your changes? Feel free to create a new PR if you find it easier to copy your changes into a new branch.
> 
> Let me know if you need help with this.

Thanks @jsoriano - I have created a new PR #23766 to clean this up.  If you could tag/review/kick jenkins off, that'd be great.  It did pass local testing. Thanks again for your assistance!

hitchatwork commented:
Opened new PR #23766 to offer cleaner review.  Closing this one out.
